language-plutus-core/src/Language/PlutusCore/Lexer/Type.hs

```diff
@@ -149,7 +149,10 @@ data Token a = LexName { loc        :: a
              deriving (Show, Eq, Generic, NFData)
 
 asBytes :: Word8 -> Doc a
-asBytes = Text 2 . T.pack . ($ mempty) . showHex
+asBytes x = Text 2 $ T.pack $ addLeadingZero $ ($ mempty) $ showHex x
+    where addLeadingZero
```

Comment:
Worth putting a type-signature on this for a bit , it took me a little while to realise how it was well-typed!